src/v2/style-guide/index.md

```diff
@@ -0,0 +1,1802 @@
+#### Bad
+
+``` html
+<input
+  v-bind:value="newTodoText"
+  :placeholder="newTodoInstructions"
+>
+```
+
+``` html
+<input
+  v-on:input="onInput"
+  @focus="onFocus"
+>
+```
+{% raw %}</div>{% endraw %}
+
+{% raw %}<div class="style-example example-good">{% endraw %}
+#### Good
+
+``` html
+<input
+  :value="newTodoText"
+  :placeholder="newTodoInstructions"
+>
+```
+
+``` html
+<input
+  v-bind:value="newTodoText"
+  v-bind:placeholder="newTodoInstructions"
+>
+```
+
+``` html
+<input
+  @input="onInput"
+  @focus="onFocus"
+>
+```
+
+``` html
+<input
+  v-on:input="onInput"
+  v-on:focus="onFocus"
+>
+```
+{% raw %}</div>{% endraw %}
+
+
+
+## Priority C Rules: Recommended (Minimizing Arbitrary Choices and Cognitive Overhead)
+
+
+
+### Component/instance options order <sup data-p="c">recommended</sup>
+
+**Component/instance options should be ordered consistently.**
+
+This is the default order we recommend for component options. They're split into categories, so you'll know where to add new properties from plugins.
+
+1. **Side Effects** (triggers effects outside the component)
+  - `el`
+
+2. **Global Awareness** (requires knowledge beyond the component)
+  - `name`
+  - `parent`
+
+3. **Component Type** (changes the type of the component)
+  - `functional`
+
+4. **Template Modifiers** (changes the way templates are compiled)
+  - `delimiters`
+  - `comments`
+
+5. **Template Dependencies** (assets used in the template)
+  - `components`
+  - `directives`
+  - `filters`
+
+6. **Composition** (merges properties into the options)
+  - `extends`
+  - `mixins`
+
+7. **Interface** (the interface to the component)
+  - `inheritAttrs`
+  - `model`
+  - `props`/`propsData`
+
+8. **Local State** (local reactive properties)
+  - `data`
+  - `computed`
+
+9. **Events** (callbacks triggered by reactive events)
+  - `watch`
+  - Lifecycle Events (in the order they are called)
+
+10. **Non-Reactive Properties** (instance properties independent of the reactivity system)
+  - `methods`
+
+11. **Rendering** (the declarative description of the component output)
+  - `template`/`render`
+  - `renderError`
+
+
+
+### Element attribute order <sup data-p="c">recommended</sup>
+
+**The attributes of elements (including components) should be ordered consistently.**
+
+This is the default order we recommend for component options. They're split into categories, so you'll know where to add custom attributes and directives.
+
+1. **Definition** (provides the component options)
+  - `is`
+
+2. **List Rendering** (creates multiple variations of the same element)
+  - `v-for`
+
+2. **Conditionals** (whether the element is rendered/shown)
+  - `v-if`
+  - `v-else-if`
+  - `v-else`
+  - `v-show`
+  - `v-cloak`
+
+3. **Render Modifiers** (changes the way the element renders)
+  - `v-pre`
+  - `v-once`
+
+4. **Global Awareness** (requires knowledge beyond the component)
+  - `id`
+
+5. **Unique Attributes** (attributes that require unique values)
+  - `ref`
+  - `key`
```

Comment:
Should `key` be together with `v-for` because of their correlation？

Reply:
No, `key` isn't specific to `v-for`, but is used in a lot of contexts where uniquely identifying the element is useful.

Reply:
OK that makes sense. I'm thinking of putting them together would make it easier to check for `key` after we created a `v-for`.